Src/AutoFakeItEasy/AutoFakeItEasyCustomization.cs

```diff
@@ -41,11 +63,40 @@ public void Customize(IFixture fixture)
         {
             if (fixture == null) throw new ArgumentNullException(nameof(fixture));
 
+            if (this.GenerateDelegates)
+            {
+                fixture.Customizations.Add(new FakeItEasyRelay(new IsDelegateSpecification()));
+            }
+
             fixture.Customizations.Add(
                 new FakeItEasyBuilder(
                     new MethodInvoker(
                         new FakeItEasyMethodQuery())));
             fixture.ResidueCollectors.Add(this.Relay);
         }
+
+        private static void AssertFakeItEasyCanFakeDelegates()
```

Comment:
The cool fail-fast approach, thanks! I completely agree with this way, as we cannot upgrade dependency for now.

Probably, it makes sense to register an issue for v5 to upgrade the min version (e.g. to v2) and remove all the legacy stuff we have.

Reply:
Done. #996